linker/shared/src/main/scala/org/scalajs/linker/frontend/IRLoader.scala

```diff
@@ -60,90 +62,19 @@ final class IRLoader(checkIR: Boolean) extends Analyzer.InputProvider
 
   def classesWithEntryPoints(): Iterable[ClassName] = entryPoints
 
-  def loadInfo(className: ClassName)(
-      implicit ec: ExecutionContext): Option[Future[Infos.ClassInfo]] = {
-    maybeGet(className, _.classInfo)
-  }
+  def classExists(className: ClassName): Boolean =
+    classNameToFile.contains(className)
 
-  def loadClassDefAndVersion(className: ClassName)(
-      implicit ec: ExecutionContext): Future[(ClassDef, Version)] = {
-    get(className, u => (u.classDef, u.version))
-  }
+  def classVersion(className: ClassName): Version =
+    classNameToFile(className).version
 
   def loadClassDef(className: ClassName)(
       implicit ec: ExecutionContext): Future[ClassDef] = {
-    get(className, _.classDef)
-  }
-
-  private def get[T](className: ClassName, f: ClassDefAndInfoCache.Update => T)(
-      implicit ec: ExecutionContext): Future[T] = {
-    maybeGet(className, f).getOrElse {
-      throw new AssertionError(s"Cannot load file for class $className")
-    }
-  }
-
-  private def maybeGet[T](className: ClassName, f: ClassDefAndInfoCache.Update => T)(
-      implicit ec: ExecutionContext): Option[Future[T]] = {
-    classNameToFile.get(className).map { irFile =>
-      cache.getOrElseUpdate(className, new ClassDefAndInfoCache)
-        .update(irFile, logger, checkIR).map(f)
-    }
+    classNameToFile(className).tree
   }
 
   def cleanAfterRun(): Unit = {
     classNameToFile = null
     entryPoints = null
-    logger = null
-    cache.filterInPlace((_, fileCache) => fileCache.cleanAfterRun())
-  }
-}
-
-private object ClassDefAndInfoCache {
-  final class Update(
-      val classDef: ClassDef,
```

Comment:
Seems we currently cache the entire ClassDef, even though the global IRCache already does that.